src/test/ui/unreachable/unwarned-match-on-never.rs

```diff
@@ -0,0 +1,18 @@
+#![deny(unreachable_code)]
+#![allow(dead_code)]
+
+#![feature(never_type)]
+
+fn foo(x: !) -> bool {
+    // Explicit matches on the never type are unwarned.
+    match x {}
+    // But matches in unreachable code are warned.
+    match x {} //~ ERROR: unreachable expression
+}
+
+fn main() {
+    return;
+    match () { //~ ERROR: unreachable expression
```

Comment:
It would be nice to test 

```rust
fn main() {
  match (return) {
    () => ()
  }
}
```

too (this should warn, I think, on the arm?)

Reply:
At the moment, this won't warn at all. I'll try to get the arms warning.